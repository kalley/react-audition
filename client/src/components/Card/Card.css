.Card {
  border: 1px solid #309bf8;
  color: #309bf8;
  margin-bottom: 20px;
  max-width: 725px;
}

.Card-Caret {
  transform: rotate(180deg);
}

.Card--collapsed .Card-Caret {
  transform: rotate(0deg);
}

.Card-title {
  align-items: center;
  appearance: none;
  background: none;
  border: 0;
  color: inherit;
  cursor: pointer;
  display: flex;
  font-size: 1.2rem;
  font-weight: 400;
  justify-content: space-between;
  margin: 0;
  padding: 34px 30px 34px 45px;
}

.Card-body {
  padding: 0 30px 34px 45px;
}
