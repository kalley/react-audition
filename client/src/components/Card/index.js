// @flow
export { default, RenderTitle, Title } from './Card';
