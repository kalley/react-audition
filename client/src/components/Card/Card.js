// @flow
import './Card.css';
import React, { type ComponentType, type Node } from 'react';
import classnames from 'classnames';
import Caret from '../Caret';

type Props = {
  children?: Node,
  title: Node,
  titleComponent: ComponentType<{ children: Node, className: string }>
};

const Card = ({ children, title, titleComponent: Title }: Props) => (
  <div
    className={classnames('Card', {
      'Card--collapsed': !children
    })}
  >
    <Title className="Card-title">
      {title}
      <Caret className="Card-Caret" size={10} />
    </Title>
    {children && <div className="Card-body">{children}</div>}
  </div>
);

Card.defaultProps = {
  titleComponent: 'h2'
};

export default Card;
