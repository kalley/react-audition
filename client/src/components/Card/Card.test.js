import React from 'react';
import { mount } from 'enzyme';
import * as emotion from 'emotion';
import { createSerializer } from 'jest-emotion';
import Card from './Card';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('<Card />', () => {
  it('renders', () => {
    const component = mount(<Card title="Test" />);

    expect(component).toMatchSnapshot();
  });

  it('renderTitle renders a different title component', () => {
    const renderTitle = ({ title }) => <div>{title}</div>;
    const component = mount(<Card renderTitle={renderTitle} title="Test" />);

    expect(component).toMatchSnapshot();
  });
});
