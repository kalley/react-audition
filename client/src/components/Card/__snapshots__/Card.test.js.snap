// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renderTitle renders a different title component 1`] = `
.emotion-0 {
  border: 1px solid #309bf8;
  color: #309bf8;
  margin-bottom: 1.25rem;
  max-width: 725px;
}

<Card
  renderTitle={[Function]}
  title="Test"
>
  <Styled(div)>
    <div
      className="emotion-0"
    >
      <div>
        Test
      </div>
    </div>
  </Styled(div)>
</Card>
`;

exports[`<Card /> renders 1`] = `
.emotion-1 {
  border: 1px solid #309bf8;
  color: #309bf8;
  margin-bottom: 1.25rem;
  max-width: 725px;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: 0;
  color: inherit;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.2rem;
  font-weight: 400;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0;
  padding: 2rem 2rem 2rem 3rem;
}

<Card
  renderTitle={[Function]}
  title="Test"
>
  <Styled(div)>
    <div
      className="emotion-1"
    >
      <Styled(h2)>
        <h2
          className="emotion-0"
        >
          Test
        </h2>
      </Styled(h2)>
    </div>
  </Styled(div)>
</Card>
`;
